src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -230,35 +262,41 @@ const StudentApplication = ({ userProfile, authentication }) => {
         )
       ) {
         // Display an error if the selected user is already in the list
-        setSnackbarText(String(newApplicantProfile.fullName) + ' is already in the list.');
-        setSnackbarSeverity('info');
-      } else if (isApplicantValid(newApplicantObject)) {
-        // Add the profile object to the list of applicants
-        setApplicationDetails((prevApplicationDetails) => ({
-          ...prevApplicationDetails,
-          Applicants: [...prevApplicationDetails.Applicants, newApplicantObject],
-        }));
-        setUnsavedChanges(true);
-        return;
+        createSnackbar(String(newApplicantProfile.fullName) + ' is already in the list.', 'info');
+      } else {
+        let validApplicant = await isApplicantValid(newApplicantObject);
+        if (validApplicant) {
+          // Add the profile object to the list of applicants
+          setApplicationDetails((prevApplicationDetails) => ({
+            ...prevApplicationDetails,
+            Applicants: [...prevApplicationDetails.Applicants, newApplicantObject],
+          }));
+          setUnsavedChanges(true);
+        }
       }
     } catch (error) {
-      setSnackbarText('Something went wrong while trying to add this person. Please try again.');
-      setSnackbarSeverity('error');
+      createSnackbar(
+        'Something went wrong while trying to add this person. Please try again.',
+        'error',
+      );
+      console.log(error);
     }
-    setSnackbarOpen(true);
   };
 
   /**
    * Callback for changing the application editor
    * @param {StudentProfileInfo} profile The StudentProfileInfo object for the person who is to be made the application editor
    */
   const handleChangeEditor = (profile) => {
-    if (profile) {
+    if (canEditApplication && profile) {
       if (
         applicationDetails.Applicants.some(
           (applicant) => applicant.Profile.AD_Username === profile.AD_Username,
         )
       ) {
+        if (unsavedChanges) {
+          saveApartmentApplication(applicationDetails);
+        }
```

Comment:
Does this mean that all changes get saved without the user knowing when they switch editors?

Reply:
Yes, that is true. I think it may have been a workaround for something, I would want to do a lot more extensive testing both with and without that line before removing that

Reply:
Presumably it's because any unsaved changes would be lost otherwise, since the user would no longer be the editor and so couldn't change anything. I would just want them to be told, explicitly, that changing editors will save all unsaved changes.

Reply:
I meant that it was an issue with causing the entire changeEditor operation to fail in some way, I forget the details of what the error was.

If you look at the handleChangeEditorAccepted, you will see that the intent is to use `saveApplication` instead of `changeApplicationEditor`, so this save behavior would happen anyways

Reply:
I agree we should have the UI text explicitly state whether the unsaved changes will be saved or not saved when changing the editor